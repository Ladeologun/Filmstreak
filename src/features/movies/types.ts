import { ImageSourcePropType } from "react-native"

export interface Movie {
    id:number
    backPoster?: string
    popularity:number
    title:string
    poster: string
    overview?: string
    rating:number
    language:string
    video?:boolean
    date:string
}


export interface TMDBMovie {
    id:number
    backdrop_path?: string
    popularity:number
    title:string
    poster_path?: string
    overview?: string
    vote_average:number
    original_language:string
    video?:boolean
    release_date: string
}

export interface NavigationTypes {
    navigate: (name: string, params?: { [key: string]: any }) => void;
    replace?: (name: string, params?: { [key: string]: any }) => void;
    canGoBack?: () => boolean;
    goBack: () => void;
}
