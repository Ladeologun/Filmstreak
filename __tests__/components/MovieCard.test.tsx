import renderer from 'react-test-renderer';
import {test,expect} from '@jest/globals';
import { MovieCard } from "~components";
import { Movie } from '~movies/types';

const MovieBuilder:Movie = {
    id: 1,
    backPoster: "https://example.com/backposter.jpg",
    popularity: 85.2,
    title: "The Great Adventure",
    poster: "https://example.com/poster.jpg",
    overview: "A thrilling adventure of a lifetime.",
    rating: 8.5,
    language: "English",
    video: true,
    date: "2024-04-12"
};

test('renders correctly', () => {
    const tree = renderer.create(<MovieCard item={MovieBuilder}  section='Popular'/>).toJSON();
    expect(tree).toMatchSnapshot();
});
