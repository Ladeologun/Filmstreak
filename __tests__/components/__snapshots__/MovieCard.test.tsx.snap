// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "marginBottom": 10,
      "width": 375,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "alignItems": "center",
        "borderRadius": 10,
        "height": 150,
        "justifyContent": "center",
        "marginBottom": 10,
        "overflow": "hidden",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://example.com/poster.jpg",
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": 150,
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <Image
      source={
        {
          "testUri": "../../../src/assets/drawables/play-button.png",
        }
      }
      style={
        {
          "height": 45,
          "tintColor": "skyblue",
          "width": 45,
        }
      }
    />
  </View>
  <Text
    numberOfLines={1}
    style={
      {
        "color": "#ffffff",
        "fontFamily": "Raleway-BoldItalic",
        "fontSize": 16,
      }
    }
  >
     
    The Great Adventure
     
  </Text>
</View>
`;
